templates/pkg/resource/sdk.go.tpl

```diff
@@ -273,3 +273,55 @@ func (rm *resourceManager) terminalAWSError(err error) bool {
 	return false
 {{- end }}
 }
+
+{{- if .CRD.GetImmutableFields }}
+// validateImmutableFields validates the immutable fields and set appropriate condition
+func (rm *resourceManager) validateImmutableFields(ko *svcapitypes.{{ .CRD.Names.Camel }},
```

Comment:
Also, since this isn't actually validating anything (and is really just identifying changes to immutable fields and adding a condition to the resource), I'd rename this function to `handleImmutableFieldsChangedCondition` or something like that...